lib/Frontend/CompilerInvocation.cpp

```diff
@@ -633,6 +643,27 @@ static bool ParseLangArgs(LangOptions &Opts, ArgList &Args,
                      A->getAsString(Args), A->getValue());
     }
   }
+  
+  // Check if the locale code is supported.
+  if (Arg *A = Args.getLastArg(OPT_locale)) {
+    std::string localeCode = A->getValue();
+    
+    auto result = llvm::find_if(supportedLocaleCodes, [&](const char * locale) {
+      return localeCode == locale;
+    });
+    
+    if (result == std::end(supportedLocaleCodes)) {
+      std::string validLocaleCodes = "";
+      for (auto code: supportedLocaleCodes) {
+        validLocaleCodes += code;
+        validLocaleCodes += ", ";
+      }
```

Comment:
I think it would be better to add them to a `std::vector` and use `llvm::join` to build a comma separated string.

Reply:
Build on what @theblixguy is saying `llvm::interleave` fits perfectly here.

Reply:
Thanks for the suggestion. I’ll modify it. :)